bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -1027,7 +1027,7 @@ impl Client {
         &self,
         room_id: String,
         via_servers: Vec<String>,
-    ) -> Result<RoomPreview, ClientError> {
+    ) -> Result<Arc<RoomPreview>, ClientError> {
```

Comment:
Do we still need all the `Arc`s here? (I'm not very sure about it myself, but I suspect we don't since `RoomPreview` is now a uniffi object)